### PR TITLE
Build Related : rename constants.json to constants.js

### LINKED ISSUE #10829: Clean up `constants.json`
## Type of issue
Improvement

## Description

[`constants.json`](https://github.com/prebid/Prebid.js/blob/master/src/constants.json) is included in every bundle, but its contents are not always needed:

 - A good portion of it is used only for native and should be excluded when native is disabled
 - `FLOOR_VALUES` is used exclusively by the pubmatic analytics adapter

### PR DESCRIPTION
**Pull request by muuki88**
## Type of change

- [x] Build related changes

## Description of change

Rename `constants.json` to `constants.js`

TODOs left

- [x] export separate consts
- [ ] if something imported only by one module, it should be moved into that module
- [ ] if something's imported only by modules, it should be moved into libraries

## Other information

- #10829

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -91,7 +91,7 @@ function lint(done) {
     '!plugins/**/node_modules/**',
     './*.js'
   ], { base: './' })
-    .pipe(eslint({ fix: !argv.nolintfix, quiet: !(typeof argv.lintWarnings === 'boolean' ? argv.lintWarnings : true) }))
+    .pipe(eslint({ fix: !argv.nolintfix, quiet: argv.lintWarnings === 'false' }))
```
Comment (muuki88) [inline gulpfile.js:94]: This setting wasn't working as `lintWarnings` was always of type `string`.

```bash
gulp lint --lintWarnings=false
```

now works as expected

**Comment by muuki88** [comment]
```
'Uncaught exception:', TypeError: Cannot read properties of undefined (reading 'AUCTION_INIT')
```

nooo

File: gulpfile.js (hunk 1 above)
Comment (dgirardi) [reply gulpfile.js:94]: it's a boolean if you use `--no-lint-warnings`.

**Comment by muuki88** [comment]
Running a single test works

```bash
gulp test --file "test/spec/modules/yieldoneBidAdapter_spec.js" --nolint
```

running all tests, it fails. yyyyyyyyyyyyyy .... @dgirardi help plz

File: src/constants.js [hunk 2]
```diff
@@ -0,0 +1,196 @@
+export default {
```
Comment (dgirardi) [inline src/constants.js:1]: I think this is the same as JSON for the minifier. It wouldn't be able to remove dead definitions; I'm curious and I'd bet it also wouldn't fix the problem you were having. 

If it doesn't, what we want is 

```
export const EVENTS = {...}; 
export const TARGETING_KEYS = {...};
...
```

or even better would be 

```
export const EVENT_AUCTION_INIT = 'auctionInit';
...
export const TARGETING_KEY_BIDDER = 'hb_bidder';
...
```

but of course the latter is a lot more work. I'm sorry I didn't make this clearer earlier.

**Comment by muuki88** [comment]
I found the issue why `constants` is not defined. I messed up some imports

File: src/constants.js [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+  PL_SIZE: 'sizes',
+  PL_BIDS: 'bids',
+  BD_BIDDER: 'bidder',
+  BD_ID: 'paramsd',
+  BD_PL_ID: 'placementId',
+  ADSERVER_TARGETING: 'adserverTargeting',
+  BD_SETTING_STANDARD: 'standard'
+};
+
+export const FLOOR_SKIPPED_REASON = {
```
Comment (dgirardi) [inline src/constants.js:12]: this should be in `priceFloors`

File: src/constants.js [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+  RANDOM: 'random'
+};
+
+export const DEBUG_MODE = 'pbjs_debug';
+
+export const STATUS = {
+  GOOD: 1
+};
+
+export const CB = {
```
Comment (dgirardi) [inline src/constants.js:23]: this appears to be unused

File: src/constants.js [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+export const NATIVE_KEYS_THAT_ARE_NOT_ASSETS = [
+  'privacyIcon',
+  'clickUrl',
+  'sendTargetingKeys',
+  'adTemplate',
+  'rendererUrl',
+  'type'
+];
+
+export const FLOOR_VALUES = {
```
Comment (dgirardi) [inline src/constants.js:199]: this should be in `pubmaticAnalyticsAdapter`

File: src/constants.js (hunk 3 above)
Comment (muuki88) [reply src/constants.js:12]: Move :+1:

File: src/constants.js (hunk 4 above)
Comment (muuki88) [reply src/constants.js:23]: Removed :wastebasket:

File: src/constants.js (hunk 5 above)
Comment (muuki88) [reply src/constants.js:199]: Move :truck:

**Comment by GeneGenie** [comment]
This is such a win for my current research, i think i could have branched from here and go fully on node.

**Comment by muuki88** [comment]
Awesome 🥳

Is there anything I can do to push this? Otherwise updating and resolving merge conflicts becomes cumbersome.

And we hit yet another issue with a bidder that doesn't work since the PUC refactoren, which is probably caused by some webpack minification, trimming away actually required constants 🫠

**Review by ChrisHuie — APPROVED** [review]

**Comment by muuki88** [comment]
Thanks @ChrisHuie 
thanks

**Pull request merged**